pkg/aggregator/tags/store.go

```diff
@@ -30,8 +31,11 @@ func (e *Entry) Tags() []string {
 
 // Release decrements internal reference counter, potentially marking
 // the entry as unused.
+//
+// Can be called concurrently with Insert and Shrink.
 func (e *Entry) Release() {
-	e.refs--
+	const minusOne = ^uint64(0)
+	atomic.AddUint64(&e.refs, minusOne)
```

Comment:
I suspect that switching to atomic operations here is going to cause a lot more bus traffic and make this quite a bit slower.  How has this fared on the benchmark env?

Reply:
No change, this doesn't happen frequently enough to make a noticeable difference. Even on a metric churn test this doesn't show up in the cpu profiles at all.

Reply:
I think you'd see it in overall CPU usage - the atomic operation will clear caches in all CPUs, and cause increased bus contention, so you'd see a general slowdown.  But if you've not seen that sort of result, great!

Reply:
My understanding is that atomics do have such drastic effects, at least on modern x64, and in most cases can be performed without any penalty at all.